src/proof/nova.rs

```diff
@@ -1912,4 +1914,39 @@ mod tests {
             false,
         );
     }
+
+    #[test]
+    fn outer_prove_begin() {
+        let mut s = Store::<Fr>::default();
+
+        let src = "(begin (emit 1) (emit 2) (emit 3))";
+
+        let expr = s.read(src).unwrap();
+        let limit = 300;
+
+        let (
+            IO {
+                expr: _result_expr,
+                env: _new_env,
+                cont: _continuation,
+            },
```

Comment:
Since these are all ignored, I don't think you need to destructure the `IO`. You could just ignore the whole thing.